babybuddy/views.py

```diff
@@ -251,7 +253,15 @@ class UserAddDevice(LoginRequiredMixin, View):
     qr_code_template = "babybuddy/login_qr_code.txt"
 
     def get(self, request):
-        qr_code_response = render(request, self.qr_code_template)
+        session_cookies = {}
```

Comment:
Lets also get some documentation here about what this is doing and why.

Also since we are unsure of this approach... are there any security concerns here?

Reply:
> Documentation

:+1:

> Also since we are unsure of this approach... are there any security concerns here?

Good question. And honest answer: I am not sure. I think the `ingress_session` is specific to the ingress-service of home assistant. If someone uses it, the session needs to come into existence somehow and something needs to be cached. This is certainly not the user's password. It might even be the case that this is a very specific key that is specific to the home assistant app, which I think would be the desired approach.

Unfortunately, the home assistant documentation of the ingress service seems to be quite "thin" and the codebase seems to be very very opaque.

Personal opinion: Since the ingress service requires a valid home assistant user session, represented by the ingress_session as far as I can tell, this seems to be a requirement for accessing home assistant in any case. It is not the password, it is specific to the ingress service. I think this is probably ok... worst thing that I experienced is that those session tokens timeout and become unusable. There might be a better way using the persistent API-tokens for home assistant, but those give access to more home assistant features...

Conclusion: :shrug: I am not quite sure, I think this is "ok" but maybe needs to be replaced or complemented in the future...

Reply:
> Unfortunately, the home assistant documentation of the ingress service seems to be quite "thin" and the codebase seems to be very very opaque.

Yeah I'm not able to find much on this, either. I'm wondering about session hijacking here but lacking any clear information or documentation it's hard to judge. I'm OK to leave it.

Reply:
Coming back to the documentation mainly:

I added some, but I feel rather than adding docs to `babybuddy/views.py` a doc-page might be even more suited. Right now, there is the qr code, but there is no information anywhere how a third-party developer can use it to authenticate with babybuddy. So, maybe a new doc-category aimed at app developers might be more fitting, which describes the json-structure in the qr-code and how to use it?

Reply:
That makes sense but deserves a separate issue, I think. I have created #674 for it.